[Zachary Chua] iP

Zacchua opened the pull request:
# Duke
> "Your mind is for having ideas, not holding them." - David Allen ([source](https://dansilvestre.com/productivity-quotes/))

Duke frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~Fast~~  *SUPER* FAST to use

All you need to do is, 
1. download it from [here](https://github.com/Zacchua/ip/releases/tag/A-Jar)
2. double-click it
3. add your tasks
4. let it manage your tasks for you 😉

And it is **FREE**!

Features
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too! Here's the `main` method:
```Java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Zacchua committed:
Level-1

Zacchua committed:
Level-2

Zacchua committed:
Level 3

Zacchua committed:
Added Todo and Deadlines (part of level-4)

Zacchua committed:
Level-4

Zacchua committed:
A-TextUiTesting

However there seems to be an issue with the first few lines, so it keeps
failing despite it looking exactly the same

Zacchua committed:
Added error handling for task adding

Implemented DukeException and DukeTaskDetailsException for handling
errors pertaining to task creation, such as not providing task name, or
task deadline for Deadlines and event timings for Events

Zacchua committed:
Level-5

Added checking for 'done' command inputs

Zacchua committed:
Level-6

Implement delete functionality. Added some javadocs, updated input.txt for testing.
Deleted unused DukeDoneInputException.java file.

Zacchua committed:
Update indentation to follow style guide

Zacchua committed:
Add save to file function (Level-7)

Saves to file data/tasks.txt in root project folder

Zacchua committed:
Add support for Date and Time (Level-8)

Zacchua committed:
Merge branch 'branch-level-7'

Zacchua committed:
Merge branch 'branch-level-8'

Zacchua committed:
Add javadocs and clean up styling

Zacchua committed:
Add more OOP

added Tasklist storage and ui classes

Zacchua committed:
A-MoreOOP

add Parser and Ui classes

Zacchua committed:
A-Packages

Zacchua committed:
A-JUnit

Add JUnit tests for todo, deadline, event, tasklist

Zacchua committed:
Add manifest for jar file (A-Jar)

Zacchua committed:
Add javadoc and fix style and indentation

Zacchua committed:
Fix styling and indentation (A-CodeStandard)

Zacchua committed:
Add find command

Zacchua committed:
Merge branch 'branch-A-JavaDoc'

Zacchua committed:
Merge branch 'branch-A-CodingStandard'

Zacchua committed:
Merge branch 'branch-Level-9'

Zacchua committed:
Merge branch 'add-gradle-support' into A-Gradle

Zacchua committed:
Add gradle support for running and unit testing

Zacchua committed:
Merge branch 'A-Gradle'

Zacchua committed:
Add JavaFX up to tutorial 2

Zacchua committed:
Add .project file

Zacchua committed:
Add .project, .classpath, .settings files

Zacchua committed:
Change from CLI to JavaFX

stanley-1 commented on src/test/java/DukeTest.java:
```diff
@@ -0,0 +1,64 @@
+import duke.exception.DukeIndexInputException;
+import duke.exception.DukeTaskDetailsException;
+import duke.task.*;
```
Would be better to list all imported classes from duke.task explicitly.

stanley-1 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,78 @@
+        duke.run();
+    }
+
+    private void run() {
```
Though JavaDoc comments aren't required for private methods, you can consider including a header comment for non-trivial private methods to aid your future reference too. 😄

stanley-1 commented on src/main/java/duke/exception/DukeException.java:
```diff
@@ -0,0 +1,20 @@
+package duke.exception;
+
+/**
+ * class for representing all exceptions pertaining to Duke
```
Try to observe capitalisation and punctuation rules for JavaDoc comments by capitalising the first letter of each sentence and adding a fullstop behind each parameter description. [Java Coding Standard: Comments]

stanley-1 commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,135 @@
+
+    /**
+     * Constructor for task that does not specify whether task is done
+     * Sets done to false by default
```
Remember to include an empty line between description and parameter section.

stanley-1 commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,135 @@
+     * @param done whether task is done or not
+     * @param deadline deadline of task
+     */
+    public Deadline(String title, Boolean done, String deadline) {
```
Boolean variables/methods can be renamed to sound like booleans. In this case, you can consider using "isDone". Same comment applies to subsequent classes of Tasks that contain this boolean variable.

stanley-1 commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,94 @@
+
+    
+    /**
+     * Method for generate the string corresponging to Command.INDEXCOMMAND
```
Rename to: "Method for generating string corresponding to Command.INDEXCOMMAND"

stanley-1 reviewed (commented):
Hi Zachary! Your codes were pleasant to review as you've done a great job in observing the Java Coding Standard. Your efforts in handling various types of Exceptions/Errors are commendable too. Here're some areas which you can consider implementing/modifying.

Zacchua committed:
Fix minor ui issues and exit command

Program exits now on exit command. Previously list commands were also
cut off if there was not enough space in the vbox, but that has been fixed.

Zacchua committed:
Merge branch 'Level-10'

Zacchua committed:
Fix small coding standard errors

reddevil1313 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,67 @@
+    private TaskList tasks;
+    private Ui ui;
+
+    public Duke() {
```
Perhaps you could add javadocs to explain what your methods do

reddevil1313 commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,57 @@
+package duke.task;
```
Nice documentation and clean code!

reddevil1313 commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,101 @@
+            }
+            return task;
+        } catch (NumberFormatException e){
+            throw new DukeIndexInputException("Please enter index of the task");
```
I like your error messages! Clearly explaining what the user should be doing instead!

reddevil1313 commented on text-ui-test/EXPECTED.TXT:
```diff
@@ -1,7 +1,66 @@
 | |_| | |_| |   <  __/
 |____/ \__,_|_|\_\___|
 
+	____________________________
```
I like that you cover all the possible test cases and commands!

reddevil1313 reviewed (commented):
Very few errors except some mission javadocs! Good Job!

Zacchua committed:
Add assertion

Zacchua committed:
Implement SLAP for Storage class

Abstracted out the parsing of tasks.txt from the getTasks method in the
Storage class into the Parser class.

Zacchua committed:
Merge pull request #1 from Zacchua/branch-A-Assertions

A-Assertion

Zacchua committed:
Merge branch 'master' into branch-A-CodeQuality

Zacchua committed:
Merge pull request #2 from Zacchua/branch-A-CodeQuality

A-CodeQuality

Zacchua committed:
Implement B-Snooze

Allow users to update the deadline for deadline tasks and event dates
for event tasks.

Zacchua committed:
Implement A-Streams

used streams for find command in tasklist

Zacchua committed:
Merge pull request #3 from Zacchua/branch-A-Streams

Implement A-Streams

Zacchua committed:
Add screenshot for website

Zacchua committed:
Add picture to readme

Zacchua committed:
Add user guide

Zacchua committed:
Update build.gradle

Zacchua committed:
changed userInterface.png name

Zacchua committed:
Fix name of Ui.png

Zacchua committed:
Fix Ui.png name in README
